mibidata/tiff.py

```diff
@@ -64,16 +64,23 @@ def write(filename, image, sed=None, optical=None, ranges=None,
         multichannel: Boolean for whether to create a single multi-channel TIFF,
             or a folder of single-channel TIFFs. Defaults to True; if False,
             the sed and optical options are ignored.
-        write_float: If True, saves the image data as float32 values (for
-            opening properly in certain software such as Halo). Defaults to
-            False which will save the image data as uint16. Note: setting
-            write_float to True does not normalize or scale the data before
-            saving, however saves the integer counts as floating point numbers.
+        dtype: Forces the image data saved as either float or uint16. Can
+            specify `'float'` or `np.float` to force data to be saved as
+            floating point values or `'uint16'` or `np.uint16` to save as
+            unsigned 16-bit integer values. Defaults to None, which saves data
+            as its original type. Note that forcing native float image data to
+            uint16 could result in a loss of precision as values are clipped.
```

Comment:
Actually, I think the easier/cleaner way for users to specify would be using the numpy type, so if anything I would remove the string names. I included them because I know plenty of times I am working with MIBItiffs but haven't actually imported numpy in that particular file. If anything, I would say simplify and leave just `np.uint16` and `np.float32`. Do you agree?

Reply:
I'm a little torn, because while in general I agree it's best to specific types by the types themselves, in this case we have two strict types we use when writing (np.uint16 and np.float32), but we are happy to convert from broader related types if not losing data (e.g. save as uint16 from np.uint8 or np.int if in [0, 2**16), save as np.float32 if float64). And, as you point out, using the specific types requires numpy to be imported. But since this parameter only gets used when a user wants to _force_ the dtype, perhaps it's best to be explicit as to the exact type used so they can think through the consequences. I'd be more hesitant if this were a required parameter, because then having to specific exact dtype would feel like an excessive burden to the user.

So, after all that, if your suggestion is to use `np.uint16` and `np.float32`, I'm fine with that. 

One last option is that we define constants based on these, so the available options to the user are from our module, and we also use these throughout our code for simplicity.

```
FLOAT = np.float32
UINT = np.uint16
```

then dtype is one of `tiff.FLOAT, tiff.UINT`, ...

Reply:
I think I would prefer using `np.uint16` and `np.float32` so that, like you say, it's explicit to the user which dtype is being saved. I'll restrict the allowed arguments to just those.